### PR TITLE
feat : add storage_root provider function for account

### LINKED ISSUE #9643: Add storage_root provider function for account
In order to obtain a storage root in rpc, we should expose this as part of a provider function, similar to

https://github.com/paradigmxyz/reth/blob/85a4008b5adaf656c4857ffe4cb85b5c66c78360/crates/storage/storage-api/src/trie.rs#L8-L8

              we would need to expose the interface in `HashedStorage` similar to `HashedPostState::from_revert_range` that only loads storage changes for a single account. then we would be able to initialize storage root with in-memory hashed state
e.g.
```rust
let address = ();
let hashed_address = keccack256(address);
let hashed_storage = HashedStorage::from_revert_range(address, &tx, block_range)?;
let hahed_storage_sorted = hashed_storage.into_sorted();
let root = StorageRoot::new_hashed(
    &tx,
    HashedPostStateStorageCursor::new(
        DatabaseHashedStorageCursor::new(
            tx.cursor_dup_read::<tables::HashedStorages>()?,
            hashed_address,
        ),
        Some(&hashed_storage_sorted)
    ),
    hashed_address,
).root()?
```

_Originally posted by @rkrasiuk in https://github.com/paradigmxyz/reth/issues/9625#issuecomment-2238750622_

### PR DESCRIPTION
**Pull request by nkysg**
closes https://github.com/paradigmxyz/reth/issues/9643

### CODE REVIEW

File: crates/storage/provider/src/providers/state/historical.rs [hunk 1]
```diff
@@ -272,6 +275,24 @@ impl<'b, TX: DbTx> StateRootProvider for HistoricalStateProviderRef<'b, TX> {
             .state_root_with_updates(self.tx)
             .map_err(|err| ProviderError::Database(err.into()))
     }
+
+    fn storage_root_from_range(
+        &self,
+        address: Address,
+        range: RangeInclusive<BlockNumber>,
+    ) -> ProviderResult<B256> {
```
Comment (mattsse) [inline crates/storage/provider/src/providers/state/historical.rs:283]: @rkrasiuk do we need the range or a single block?

from rpc pov we need the root at a specific block

Comment (rkrasiuk) [reply crates/storage/provider/src/providers/state/historical.rs:283]: you need the revert range, but the end block should always be latest, so we can just do a single block

**Review by mattsse — CHANGES_REQUESTED** [review]
sorry about the conflicts @nkysg there's currently some movement

but otherwise this looks good

ptal @rkrasiuk

**Review by mattsse — APPROVED** [review]
lgtm, pending @rkrasiuk

**Review by rkrasiuk — APPROVED** [review]

**Pull request merged**